packages/scandipwa/src/route/ProductPage/ProductPage.container.js

```diff
@@ -193,6 +193,21 @@ export class ProductPageContainer extends PureComponent {
     }
 
     componentDidMount() {
+        const {
+            productSKU,
+            product: {
+                sku
+            }
+        } = this.props;
+
+        /**
+         * If the currently loaded category ID does not match the ID of
+         * category ID from URL rewrite, request category.
+         */
+        if (productSKU !== sku) {
+            this.requestProduct();
+        }
+
```

Comment:
What if we go from PDP to the related component page (PDP too)?
Will the `componentDidMount` be called in this case?

Reply:
Yeah, componentDidMount is called again when we go to another product in related products